src/test/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseDataTest.scala

```diff
@@ -496,10 +497,53 @@ class AppInfoBaseDataTest extends UnitTest with GroupCreation {
       f.groupManager.podVersion(any, any) returns Future.successful(None)
       f.marathonSchedulerService.listRunningDeployments() returns Future.successful(Seq.empty)
 
-      When("requesting pod status")
-      f.baseData.podStatus(pod).futureValue
+      Then("requesting pod status should not throw an exception")
+      noException should be thrownBy {
+        f.baseData.podStatus(pod).futureValue
+      }
+    }
 
-      Then("no exception was thrown so status was successfully fetched")
+    "show a pod status with a task in TASK_UNKNOWN state" in {
+      import scala.concurrent.ExecutionContext.Implicits.global
+      val f = new Fixture
+      Given("A pod instance")
+      val taskFailure = TaskFailureTestHelper.taskFailure
+      f.taskFailureRepository.get(pod.id) returns Future.successful(Some(taskFailure))
+      val instance1 = {
+        val instanceId = Instance.Id.forRunSpec(pod.id)
```

Comment:
There if a `f.fakeInstance` method but I found it to cumbersome to wire up the Mesos task condition.